File: cpp/leetcode/326.cpp
class Solution {
public:
    bool isPowerOfThree(int n) {
        if(n > 1) {
            while(n%3==0) n /= 3;
        }
        return n==1;
    }
};

class Solution {
public:
    bool isPowerOfThree(int n) {
        return fmod(log10(n)/log10(3), 1)==0;
    }
};

class Solution {
public:
    int const Max3PowerInt = 1162261467; // 3^19, 3^20 = 3486784401 > MaxInt32
    int const MaxInt32 = 2147483647; // 2^31 - 1
    bool isPowerOfThree(int n) {
        if (n <= 0 || n > Max3PowerInt) return false;
        return Max3PowerInt % n == 0;
    }
};

File: python/leetcode/326.py
class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 0: return False
        while n % 3 == 0:
            n /= 3
        return n == 1

    def isPowerOfThreeR(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0: return False
        if n == 1: return True
        return n % 3 == 0 and self.isPowerOfThreeR(n / 3)

    def isPowerOfThree(self, n):
        return n > 0 == 3**19 % n


if __name__ == '__main__':
    test = Solution()
    print test.isPowerOfThree(1)